drafts/HTTP-Response-Compression.adoc

```diff
@@ -0,0 +1,245 @@
+{bl}
+{bl}
+  For instance, consider the following header: `Accept-Encoding: deflate;q=0.5, gzip;q=1.0`
+{bl}
+{bl}
+  Without defining a server preferred encoding, the server would pick `gzip`, as it is the highest ranked quality value in the list. However, if `deflate` were to be set as the server preferred encoding algorithm, the same `Accept-Encoding` header would result in `deflate` being chosen. Find more information on this attribute on the <<Ready to Try Compression Yourself?>> section.
+{bl}
+{bl}
+. *Using the HTTP Clients preferred algorithm*
+{bl}
+The values for the `Accept-Encoding` header are sorted in descending order and each value is evaluated until a valid encoding is found. The highest ranked valid algorithm is chosen. If the `gzip` algorithm is tied among the highest ranked, it will be chosen as the algorithm to use. If there are no explicitly named encoding algorithms, and the asterisk (`{star}`) symbol is used, `gzip` will be used if able. If not able to use `gzip`, `deflate` will be used. If both are declared as unsupported by means of a quality value of `0.000`, no content encoding algorithm is chosen.
+{bl}
+{bl}
+*For example,*
+
+     - The HTTP client can specify the preferred encoding algorithm as `gzip` by giving it the highest quality value.
+
+    Accept-Encoding: deflate;q=0.5, gzip;q=1.0
+
+      - The HTTP client can specify multiple encoding algorithms with the highest quality value. In such cases, the server will pick the `gzip` algorithm if able.
+
+      Accept-Encoding: deflate;q=1.0, gzip;q=1.0
+
+      - The HTTP client may also indicate all encodings are supported. Since the `gzip` algorithm is supported, the server will choose it.
+
+      Accept-Encoding: *
+
+      - The HTTP client can specify the `gzip` algorithm is *not* supported. In this case, the `deflate` algorithm is used instead.
+
+      Accept-Encoding: gzip;q=0, *
+
+      - The HTTP client can specify it does *not* support either the `gzip` or `deflate` algorithms. In this case, no compression algorithm will be chosen.
+
+      Accept-Encoding: gzip;q=0, deflate;q=0, *
+
+
+If no content encoding algorithm was chosen by the steps above, or if the encoding algorithm is set to `identity`, *no* compression will be realized on the response.
+
+Additionally, since the `Accept-Encoding` header influences the Open Liberty's process for selecting and representing the response, a `Vary` header with the value of `Accept-Encoding` is also added to the response. This will happen regardless of whether the response is ultimately compressed or not, as long as the compression feature is configured in the `server.xml`.
```

Comment:
Give some information on the vary header we shouldn't assume everyone knows what it is / does.